EIPS/eip-restricted_chain_context_txs.md

```diff
@@ -0,0 +1,180 @@
+`segmentId` further restricts transaction application to one subsection ("segment") of one chain.
+
+From this constraint hierarchy, we note that it is possible an implementation of `segmentId` to make `chainId` redundant.
+
+##### Construction
+
+The value is generally constructed as a concatenated `BlockNumber``BlockHashPrefix`, where `BlockHashPrefix` is __the first 4 bytes of the block hash of the canonical block with number `BlockNumber`__.
+
+If the referenced `BlockNumber` is `0` it may be omitted. Otherwise, `BlockNumber` should be a __little endian encoded value with trailing zero values truncated__.
+This scheme yields a minimum of 0 bytes and a maximum of 8 bytes (uint64).
+
+Decoding can be done by chomping the last 4 bytes of the value as the required prefix of a block hash.
+The remaining bytes (those preceeding the last 4 bytes, if any) can be decoded from a little endian into a uint64 or equivalent type.
+In case of 0 bytes for the block number, 0 (genesis block) can be safely assumed.
+
+Example values are given below.
+
+| Segment ID | Block Number | Canonical Block Hash |
+| --- | --- | --- |
+| `e78b1ec3` | `0` | `0xe78b1ec31bcb535548ce4b6ef384deccad1e7dc599817b65ab5124eeaaee3e58` |
+| `01e78b1ec3` | `1` | `0xe78b1ec31bcb535548ce4b6ef384deccad1e7dc599817b65ab5124eeaaee3e58` |
+| `c0e1e4e78b1ec3` | `15_000_000` | `0xe78b1ec31bcb535548ce4b6ef384deccad1e7dc599817b65ab5124eeaaee3e58` |
+| `ff0fa5d4e8e78b1ec3` | `999_999_999_999` | `0xe78b1ec31bcb535548ce4b6ef384deccad1e7dc599817b65ab5124eeaaee3e58` |
+| `ffffffffffffff7fe78b1ec3` | `9223372036854775807` | `0xe78b1ec31bcb535548ce4b6ef384deccad1e7dc599817b65ab5124eeaaee3e58` |
+
+##### Encoding
+
+Encoding should follow the same rules as the optional `data` value.
+
+#### `eligibleMinerList`
+
+- `ANNOTATION_PREFIX` `2`.
+- `ELIGIBLE_MINER_LIST` `[address...]`. A list of addresses.
+- `MAX_ELEMENTS` `10`. The maximum number of addresses that can be provided.
+
+Constrains the validity of a transaction by a chain context subclass referencing the miner address of the block including
+the transaction.
+The transaction is only valid when included in a block having an `etherbase` contained in annotated list of addresses.
+
+##### Encoding
+
+Encoding should follow the same rules as the optional `accessList` value.
+
+#### `expiry`
+
+- `ANNOTATION_PREFIX` `4`. 
+
+Constrains the validity of a transaction by a chain context subclass referencing the timestamp of the block including the transaction (in seconds-since Unix epoch).
+The transaction is only valid when included in a block having a `timestamp` less than the value annotated.
+
+##### Encoding
+
+Encoding should follow the same rules as other transaction integer values, eg. `nonce`, `gasPrice`, `gasLimit`, etc.
```

Comment:
This doesn't specify where the timestamp should go.  Also, be explicit about encoding mechanism rather than referring to other fields.